src/modules/flight_mode_manager/FlightModeManager.cpp

```diff
@@ -137,7 +137,8 @@ void FlightModeManager::updateParams()
 void FlightModeManager::start_flight_task()
 {
 	// Do not run any flight task for VTOLs in fixed-wing mode
-	if (_vehicle_status_sub.get().vehicle_type == vehicle_status_s::VEHICLE_TYPE_FIXED_WING) {
+	if (_vehicle_status_sub.get().vehicle_type == vehicle_status_s::VEHICLE_TYPE_FIXED_WING ||
+	    _vehicle_status_sub.get().nav_state >= vehicle_status_s::NAVIGATION_STATE_EXTERNAL1) {
```

Comment:
Might be worth explicitly checking the range here (EXTERNAL1-EXTERNAL8) in case someone slips on a new navigation state in the future.

Reply:
I was assuming the external ones are always at the top of the range. I'd add internal ones before the external ones. But you're right, I'm happy to add that.

Reply:
I changed it interactively to only cover the external mode range.